staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -290,80 +291,83 @@ func patchResource(
 			// 3. ensure no conflicts between the two patches
 			// 4. apply the #1 patch to the currentJS object
 
-			// Since the patch is applied on versioned objects, we need to convert the
-			// current object to versioned representation first.
-			currentVersionedObject, err := unsafeConvertor.ConvertToVersion(currentObject, kind.GroupVersion())
-			if err != nil {
-				return nil, err
-			}
-			currentObjMap, err := runtime.DefaultUnstructuredConverter.ToUnstructured(currentVersionedObject)
-			if err != nil {
-				return nil, err
-			}
-
-			var currentPatchMap map[string]interface{}
-			if originalObjMap != nil {
-				var err error
-				currentPatchMap, err = strategicpatch.CreateTwoWayMergeMapPatch(originalObjMap, currentObjMap, versionedObj)
+			// Calculate only if strategic merge patch is supported.
+			if sets.NewString(supportedPatchTypes...).Has(string(types.StrategicMergePatchType)) {
```

Comment:
Only added the if clause here. Didn't change anything below this.

Reply:
I think that if it doesn't support SMP it needs to return something but I'm not sure what